<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="4" passed="2" failed="2" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2022-03-12T23:51:36 CST" name="Default suite" finished-at="2022-03-12T23:51:59 CST" duration-ms="22730">
    <groups>
    </groups>
    <test started-at="2022-03-12T23:51:36 CST" name="Default test" finished-at="2022-03-12T23:51:59 CST" duration-ms="22730">
      <class name="testcases.LoginTest">
        <test-method is-config="true" signature="DataSetUp()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:36 CST" name="DataSetUp" finished-at="2022-03-12T23:51:37 CST" duration-ms="418" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- DataSetUp -->
        <test-method is-config="true" signature="SetUp()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:37 CST" name="SetUp" finished-at="2022-03-12T23:51:39 CST" duration-ms="2037" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUp -->
        <test-method signature="ExternalDataTest()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:39 CST" name="ExternalDataTest" finished-at="2022-03-12T23:51:43 CST" duration-ms="4275" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "com.relevantcodes.extentreports.ExtentTest.log(com.relevantcodes.extentreports.LogStatus, String, String)" because "this.test" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "com.relevantcodes.extentreports.ExtentTest.log(com.relevantcodes.extentreports.LogStatus, String, String)" because "this.test" is null
at pages.LoginPage.LoginFunction(LoginPage.java:48)
at testcases.LoginTest.ExternalDataTest(LoginTest.java:55)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:567)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- ExternalDataTest -->
        <test-method is-config="true" signature="TearDown()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:43 CST" name="TearDown" finished-at="2022-03-12T23:51:43 CST" duration-ms="165" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDown -->
        <test-method is-config="true" signature="SetUp()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:43 CST" name="SetUp" finished-at="2022-03-12T23:51:45 CST" duration-ms="1447" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUp -->
        <test-method signature="LoginFailureTest()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:45 CST" name="LoginFailureTest" finished-at="2022-03-12T23:51:51 CST" duration-ms="6111" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- LoginFailureTest -->
        <test-method is-config="true" signature="TearDown()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:51 CST" name="TearDown" finished-at="2022-03-12T23:51:51 CST" duration-ms="154" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDown -->
        <test-method is-config="true" signature="SetUp()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:51 CST" name="SetUp" finished-at="2022-03-12T23:51:52 CST" duration-ms="1491" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SetUp -->
        <test-method signature="LoginSuccessTest()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:52 CST" name="LoginSuccessTest" finished-at="2022-03-12T23:51:59 CST" duration-ms="6307" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- LoginSuccessTest -->
        <test-method is-config="true" signature="TearDown()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:59 CST" name="TearDown" finished-at="2022-03-12T23:51:59 CST" duration-ms="152" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- TearDown -->
        <test-method signature="ParameterizedTest(java.lang.String,java.lang.String)[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:59 CST" name="ParameterizedTest" finished-at="2022-03-12T23:51:59 CST" duration-ms="0" status="FAIL">
          <exception class="org.testng.TestNGException">
            <message>
              <![CDATA[
Parameter 'Param1' is required by @Test on method ParameterizedTest but has not been marked @Optional or defined
in C:\Users\Sahosa02\AppData\Local\Temp\testng-eclipse--246890611\testng-customsuite.xml]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.testng.TestNGException: 
Parameter 'Param1' is required by @Test on method ParameterizedTest but has not been marked @Optional or defined
in C:\Users\Sahosa02\AppData\Local\Temp\testng-eclipse--246890611\testng-customsuite.xml
at org.testng.internal.Parameters.createParams(Parameters.java:270)
at org.testng.internal.Parameters.createParametersForMethod(Parameters.java:358)
at org.testng.internal.Parameters.createParameters(Parameters.java:700)
at org.testng.internal.Parameters.handleParameters(Parameters.java:884)
at org.testng.internal.Parameters.handleParameters(Parameters.java:740)
at org.testng.internal.ParameterHandler.handleParameters(ParameterHandler.java:59)
at org.testng.internal.ParameterHandler.createParameters(ParameterHandler.java:38)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:791)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- org.testng.TestNGException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- ParameterizedTest -->
        <test-method is-config="true" signature="DataTearDown()[pri:0, instance:testcases.LoginTest@f736069]" started-at="2022-03-12T23:51:59 CST" name="DataTearDown" finished-at="2022-03-12T23:51:59 CST" duration-ms="149" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- DataTearDown -->
      </class> <!-- testcases.LoginTest -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
